c/cobuild.c

```
@@ -879,23 +744,22 @@ int ckb_cobuild_entry(ScriptEntryType callback, bool *cobuild_enabled) {
   hs = otx_start.start_header_deps;
   he = hs;
   printf("ie = %d, oe = %d, ce = %d, he = %d", ie, oe, ce, he);
-  size_t index = i + 1;
+  uint32_t index = i + 1;
   printf("Otx starts at index %d(inclusive)", index);
-  for (;; index++) {
-    mol2_cursor_t cursor;
-    err = ckb_new_witness_cursor(&cursor, g_cobuild_seal_data_source,
-                                 MAX_CACHE_SIZE, index, CKB_SOURCE_INPUT);
-    // step 6, not WitnessLayoutOtx
-    CHECK_LOOP(err);
-    uint32_t id = 0;
-    err = try_union_unpack_id(&cursor, &id);
-    if (err || id != WitnessLayoutOtx) {
+  for (; index < witness_len; index++) {
+    WitnessLayoutType witness_layout = {0};
+    err = get_witness_layout(witnesses, index, &witness_layout);
+    if (err != 0) {
```

Use CHECK_LOOP, in case there is any error inside get_wintess_layout

---

Note we are only doing iteration here within `witness_len`, right now no syscall would throw `CKB_OUT_OF_BOUND` error, meaning `CHECK_LOOP` does not apply here.

Any errors returned by `get_witness_layout` will be essentially due to misformed witness format, personally I think we can safely ignore errors from `get_witness_layout` here.